plugins/terminal.koplugin/main.lua

```diff
@@ -78,10 +79,10 @@ function Terminal:execute()
         buttons_table = {
             {
                 {
-                    text = 'New command(s)...',
+                    text = _("Back"),
                     callback = function()
                         UIManager:close(viewer)
-                        Terminal:onTerminalStart()
+                        self.ui:handleEvent(Event:new("TerminalStart"))
```

Comment:
May be just `self:onTerminalStart()` (the instance, that remembers the input) rather than your initial `Terminal:onTerminalStart()` (the class object, which was another object), was just enough.

Reply:
You're right, thanks! I've modified the code accordingly.